### DISCUSSION TITLE
Renovate is crashing (occasionly) with segmentation fault error

### DISCUSSION THREAD

**Discussion by gregory-hive** [renovatebot/renovate#30369] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, version "latest"

### Please tell us more about your question or problem

Hi there!

We are running Renovate scans for a couple of dozen repositories every morning in Azure DevOps pipelines. For 3-5 of the repositories, we get a "Segmentation fault" error every day.

I could not pinpoint exactly why this is happening, but here are some observations:
1. Segmentation fault error happens when renovate is fetching changelogs for some dependency:
```
DEBUG: Found existing PR (repository=repo, branch=renovate/node-20.x)
DEBUG: Fetching changelog: https://github.com/nodejs/node (v18.20.4 -> v20.15.1) (repository=repo, branch=renovate/node-20.x)
DEBUG: Fetching changelog: https://github.com/DefinitelyTyped/DefinitelyTyped (18.19.31 -> 20.14.10) (repository=repo, branch=renovate/node-20.x)
DEBUG: Fetching changelog: https://github.com/nodejs/node (18.20.3 -> 20.15.0) (repository=repo, branch=renovate/node-20.x)
DEBUG: Fetching changelog: https://github.com/nodejs/node (18.20.3 -> 20.15.0) (repository=repo, branch=renovate/node-20.x)
/__w/_temp/004e42df-66e1-43e5-9cc6-4938acea0f11.sh: line 3:  1696 Segmentation fault      (core dumped) RENOVATE_REPOSITORIES="$repo" npx renovate
```
2. When I run the renovate image locally on all repos without an error, but it fails when running in Azure DevOps pipeline. I can also reproduce error locally when using the same access token that the pipeline is using.
3. The repos that renovate crashes on vary a bit - sometime it's 3 repos, sometimes 4 or 5. But it happens always for the same repositories, so it is probably related to their size or specific dependencies 

Is there anything you could recommend us to try to resolve this error or can a fix be applied on Renovate's side? Thank you!

**Comment by viceice** [comment, marked as answer]
run our docker image instead of using npx, that system is probably using some incompatible libs.

maybe that system has  not enough memory to run your big repos

**Reply by gregory-hive** [reply]
Running Renovate as a Docker image has resolved segmentation fault error. 
Thank you!
